auslib/db.py

```diff
@@ -1096,6 +1139,11 @@ def addLocaleToRelease(self, name, product, platform, locale, data, old_data_ver
             raise ValueError("Release blob contains forbidden domain.")
         where = [self.name == name]
         what = dict(data=releaseBlob.getJSON())
+
+        product = self.select(where=where, columns=[self.product], transaction=transaction)[0]["product"]
```

Comment:
I think my thinking here was that it was nice to throw user input errors (ie: badly formed data) even if the user is not allowed to modify it. On second thought though, that's not really important, and consistency trumps it.